libs/stats/odc/stats/_cli_generate_mosaic.py

```diff
@@ -77,48 +81,60 @@ def save(xx, location, product_name, verbose):
         NUM_THREADS="ALL_CPUS",
         BIGTIFF="YES",
         SPARSE_OK=True,
+        creds=creds,
     )
 
 
-@click.command()
+@main.command("generate-mosaic")
 @click.argument("product", type=str)
 @click.argument("input_prefix", type=str)
 @click.argument("location", type=str)
+@click.option("--bands", type=str)
 @click.option("--verbose", "-v", is_flag=True, help="Be verbose")
-def cli(product, input_prefix, location, verbose):
+def generate_mosaic(product, input_prefix, location, bands, verbose):
     """
     Generate mosaic overviews of the stats data.
 
     An intermediate cache file is generated and stored in the output location
     during this process.
     Note: The input bucket must be public otherwise the data can not be listed.
     """
-
+    bands = bands.split(",")
     product = product_from_yaml(product)
     if verbose:
         print(f"Preparing mosaics for {product.name} product")
 
     dss = s3_fetch_dss(input_prefix, product, glob="*.json")
-    cache = create_cache(f"{product.name}.db")
-
-    if verbose:
-        print(f"Writing {location}/{product.name}.db")
-
-    cache = create_cache(f"{location}/{product.name}.db")
-    cache.bulk_save(dss)
-    if verbose:
-        print(f"Found {cache.count:,d} datasets")
 
     dc = Datacube()
-    dss = list(cache.get_all())
     xx = dc.load(
         datasets=dss,
-        dask_chunks={"x": 3200, "y": 3200},
+        dask_chunks={"x": 2048, "y": 2048},
         resolution=(-120, 120),
-        measurements=["red", "green", "blue"],
+        measurements=bands,
     )
 
-    save(xx, location, product.name, verbose)
+    if bands is None:
+        suffix = ''
+    else:
+        suffix = '_' + '_'.join(bands)
+
+    if verbose:
+        print(f"Writing {location}/{product.name}{suffix}.tif")
+    xx = xx.squeeze('time').to_stacked_array('bands', ['x', 'y'])
+    yy = save_cog(
```

Comment:
Should this be calling `save()`, not `save_cog()`?

Reply:
nah, `save` is a wrapper for `write_cog` which loads everything into memory!